controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -180,6 +239,30 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(request ctrl.Reques
 
 	enactmentConditions.NotifyMatching()
 
+	if !instance.Spec.Parallel {
+		conditionCount, err := r.getEnactmentCount(request.NamespacedName)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("error getting enactment ")
+		}
+
+		if conditionCount.Aborted() > 0 {
+			enactmentConditions.NotifyConfigurationAborted(fmt.Errorf("another nodes failed to apply configuration"))
+			return reconcile.Result{}, fmt.Errorf("another nodes failed configuration, aborting")
+		}
+
+		// TODO:
+		//if metav1.Now().Sub(instance.Status.NodeUpdateStart.Time) > nodeConfigurationTimeout {
+		//	// a node has been running the update for too long
+		//	enactmentConditions.NotifyConfigurationAborted(fmt.Errorf("failed to configure in time"))
+		//	return reconcile.Result{}, nil
+		//}
+
+		err = r.setNodeRunningUpdate(request.NamespacedName)
+		if err != nil {
+			return reconcile.Result{RequeueAfter: nodeConfigurationRequeueTime}, nil
```

Comment:
Are we sure we don't want to block `Reconcile` ? I am a little concerned about nncp changed in the middle of rollout (We are suppose to not support multiple nncps).

Reply:
I was thinking about blocking the clearing function that removes the runningNode field, I didn't realize this.
In this case, I think it makes sense to block at both setting and unsetting the runningNode field